scripts/gha/test_simulator.py

```diff
@@ -387,23 +389,77 @@ def _build_tvos_helper(helper_project, device_name, device_os):
         return os.path.join(file_dir, file_name)
 
 
-def _record_apple_tests(video_name):
-  command = "xcrun simctl io booted recordVideo -f --codec=h264 %s" % video_name
-  logging.info("Recording test: %s", command)
-  return subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True, preexec_fn=os.setsid)
-
-
-def _stop_recording(record_process):
-  logging.info("Stop recording test")
-  try:
-    os.killpg(record_process.pid, signal.SIGINT)
-  except:
-    logging.info("Stop recording test failed!!!")
+def _record_apple_tests(video_name, display):
+  args = [
+    "xcrun",
+    "simctl",
+    "io",
+    "booted",
+    "recordVideo",
+    "-f",
+    "--codec=h264",
+  ]
+  if display is not None:
```

Comment:
nit: if display:
Also, we could provide the default value "external" or "internal" (Not sure what's the difference here).

Reply:
I'd rather not use `if display` here because then it conflates the empty string with None, and, IMO, they should be treated differently.

I only add the `--display` argument to the command line if something other than the default (which is "internal") is specified. Although explicitly specifying `--display=internal` for iOS tests should work, it's not necessary.

If you feel strongly about these comments then I'm happy to discuss and I can fix in a follow-up PR.